How to run on windows?

Hello, I am having trouble trying to compile on windows10. 

`.\protoc.exe -I=proto_src --ts_out=dist message.proto`  throws `'protoc-gen-ts' is not recognized as an internal or external command, operable program or batch file. 
--ts_out: protoc-gen-ts: Plugin failed with status code 1.`  error.

And when I use `.\protoc.exe --plugin=protoc-gen-ts=<absolute_path_to_project>/node_modules/protoc-gen-ts/bin/protoc-gen-ts -I=proto_src --ts_out=dist message.proto`, it throws `--ts_out: protoc-gen-ts: %1 is not a valid Win32 application.` And it only takes absolute path. Doesn't work with relative path.

I have observed that in other typescript generators, there was cmd script given in bin folder of package. But in my npm download, there is only javascript code without any extension to the file but with first line saying '#!/usr/bin/env node'. This made me think if this is made only for shells in unix systems and not windows. 

I neither have bazel on my system and nor am I familiar with it. So, I would like to know how to generate files on windows(if it is possible), preferably without bazel.

`> protoc-gen-ts@0.3.6-rc4 test D:\open_source\protoc-gen-ts\protoc-gen-ts
> bazel build //test:codegen && node ./scripts/sync_generated_protos.js test && bazel test //test/... --test_output=errors

INFO: Repository npm instantiated at:
  D:/open_source/protoc-gen-ts/protoc-gen-ts/WORKSPACE:21:13: in <toplevel>
  C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/index.bzl:78:18: in yarn_install
Repository rule yarn_install defined at:
  C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl:644:31: in <toplevel>
ERROR: An error occurred during the fetch of repository 'npm':
   Traceback (most recent call last):
        File "C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 610, column 15, in _yarn_install_impl
                _copy_file(repository_ctx, repository_ctx.attr.package_json)
        File "C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 343, column 17, in _copy_file
                fail("mkdir -p %s failed: \nSTDOUT:\n%s\nSTDERR:\n%s" % (dirname, result.stdout, result.stderr))
Error in fail: mkdir -p _ failed:
STDOUT:

STDERR:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("mkdir" -p _): The system cannot find the file specified.      
 (error: 2)
ERROR: Error fetching repository: Traceback (most recent call last):
        File "C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 610, column 15, in _yarn_install_impl
                _copy_file(repository_ctx, repository_ctx.attr.package_json)
        File "C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 343, column 17, in _copy_file
                fail("mkdir -p %s failed: \nSTDOUT:\n%s\nSTDERR:\n%s" % (dirname, result.stdout, result.stderr))
Error in fail: mkdir -p _ failed:
STDOUT:

STDERR:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("mkdir" -p _): The system cannot find the file specified.      
 (error: 2)
ERROR: Skipping '//test:codegen': no such package '@npm//@bazel/typescript': mkdir -p _ failed:
STDOUT:

STDERR:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("mkdir" -p _): The system cannot find the file specified.      
 (error: 2)
WARNING: Target pattern parsing failed.
ERROR: no such package '@npm//@bazel/typescript': mkdir -p _ failed:
STDOUT:

STDERR:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("mkdir" -p _): The system cannot find the file specified.      
 (error: 2)
INFO: Elapsed time: 0.205s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
    currently loading: test
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! protoc-gen-ts@0.3.6-rc4 test: `bazel build //test:codegen && node ./scripts/sync_generated_protos.js test && bazel test //test/... --test_output=errors`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the protoc-gen-ts@0.3.6-rc4 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\4667r\AppData\Roaming\npm-cache\_logs\2021-06-02T05_14_18_754Z-debug.log`

I am getting this error when I am trying to run `npm test`. Unable to figure out what the problem is. I installed yarn also globally thinking that could be the problem.

I am facing this error when running `npm test`. Unable to figure out the cause. 

```
> protoc-gen-ts@0.3.6-rc4 test D:\open_source\protoc-gen-ts\protoc-gen-ts
> bazel build //test:codegen && node ./scripts/sync_generated_protos.js test && bazel test //test/... --test_output=errors

INFO: Repository npm instantiated at:
  D:/open_source/protoc-gen-ts/protoc-gen-ts/WORKSPACE:21:13: in <toplevel>
  C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/index.bzl:78:18: in yarn_install
Repository rule yarn_install defined at:
  C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl:644:31: in <toplevel>
ERROR: An error occurred during the fetch of repository 'npm':
   Traceback (most recent call last):
        File "C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 610, column 15, in _yarn_install_impl
                _copy_file(repository_ctx, repository_ctx.attr.package_json)
        File "C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 343, column 17, in _copy_file
                fail("mkdir -p %s failed: \nSTDOUT:\n%s\nSTDERR:\n%s" % (dirname, result.stdout, result.stderr))
Error in fail: mkdir -p _ failed:
STDOUT:

STDERR:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("mkdir" -p _): The system cannot find the file specified.      
 (error: 2)
ERROR: Error fetching repository: Traceback (most recent call last):
        File "C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 610, column 15, in _yarn_install_impl
                _copy_file(repository_ctx, repository_ctx.attr.package_json)
        File "C:/users/4667r/_bazel_4667rahul/vgrfovvs/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 343, column 17, in _copy_file
                fail("mkdir -p %s failed: \nSTDOUT:\n%s\nSTDERR:\n%s" % (dirname, result.stdout, result.stderr))
Error in fail: mkdir -p _ failed:
STDOUT:

STDERR:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("mkdir" -p _): The system cannot find the file specified.      
 (error: 2)
ERROR: Skipping '//test:codegen': no such package '@npm//@bazel/typescript': mkdir -p _ failed:
STDOUT:

STDERR:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("mkdir" -p _): The system cannot find the file specified.      
 (error: 2)
WARNING: Target pattern parsing failed.
ERROR: no such package '@npm//@bazel/typescript': mkdir -p _ failed:
STDOUT:

STDERR:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("mkdir" -p _): The system cannot find the file specified.      
 (error: 2)
INFO: Elapsed time: 0.205s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
    currently loading: test
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! protoc-gen-ts@0.3.6-rc4 test: `bazel build //test:codegen && node ./scripts/sync_generated_protos.js test && bazel test //test/... --test_output=errors`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the protoc-gen-ts@0.3.6-rc4 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\4667r\AppData\Roaming\npm-cache\_logs\2021-06-02T05_14_18_754Z-debug.log
```

--- a/bin/protoc-gen-ts.cmd
+++ b/bin/protoc-gen-ts.cmd
@@ -1,0 +1,14 @@
+@echo off
+
+@REM check if package is installed locally.
+for /f %%i in ('npm root') do set install_path=%%i
+
+@REM if not installed locally, set install path to global one
+setlocal enabledelayedexpansion
+if not exist !install_path!\protoc-gen-ts\  (
+    for /f %%i in ('npm root -g') do set install_path=%%i
+)
+endlocal & set install_path=%install_path%
+
+@REM invoke the index.js 
+node %install_path%\protoc-gen-ts\src\index

